src/test/java/seedu/address/logic/parser/ParserUtilTest.java

```diff
@@ -275,4 +276,28 @@ public void parseDays_collectionWithValidValue_returnsDaySet() throws Exception
                 List.of("MONDAY", "TUESDAY", "WEDNESDAY", "THURSDAY", "FRIDAY", "SATURDAY", "SUNDAY")));
     }
 
+    @Test
+    public void parseSubject_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> ParserUtil.parseSubject(null));
+    }
+
+    @Test
+    public void parseSubject_invalidValue_throwsParseException() {
+        // Invalid subject example: contains special characters or is empty
+        assertThrows(ParseException.class, () -> ParserUtil.parseSubject("Math!")); // contains invalid characters
+        assertThrows(ParseException.class, () -> ParserUtil.parseSubject("")); // empty string
+    }
+
+    @Test
+    public void parseSubject_validValueWithoutWhitespace_returnsSubject() throws Exception {
+        Subject expectedSubject = new Subject("Mathematics");
+        assertEquals(expectedSubject, ParserUtil.parseSubject("Mathematics"));
```

Comment:
Would it be better to declare a final string ValidSubject variable instead of typing out the subject "Mathematics"?

Reply:
Thank you for the suggestion. I have declared a validSubject variable.